### DISCUSSION TITLE
Storing additional list values in `trial.json`?

### DISCUSSION THREAD

**Discussion by Anselmoo** [keras-team/keras-tuner#1008] [Q&A]
Is it possible to store a `list` during the run? `oracle.update_trial` unfortunately only supports `float` or `str`.

```py
   self.oracle.update_trial(
            trial.trial_id,
            {
                "objective_1":  float or str,
                "objective_2":  float or str
                "objective_3":  float or str      
            },
```

preferred:

```py
   self.oracle.update_trial(
            trial.trial_id,
            {
                "objective_1":  [float, float, str]
            },
```

Would it be very difficult to extend `keras_tuner` by a new function, `self.oracle.update_meta,` due to the message protocol?

thx

**Comment by santi-mir** [comment]
I've [added this](https://github.com/ghsanti/keras-tuner/blob/main/keras_tuner%2Fprotos%2Fkeras_tuner.proto#L94).


It is only meant to work with new Keras though (and run multibackend wo crashing hopefully.), no backwards compatibility.

There are  "example outputs" that you can check the format of. https://github.com/ghsanti/keras-tuner

(PS: The general idea is: save all the details to `trial.json`, but provide use with optional summaries etc.)
